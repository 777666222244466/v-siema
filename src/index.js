import Siema from './components/Siema.vue';

export default {
    install(Vue, options) {
        Vue.component('siema', Siema);
    },
};

export { Siema };
