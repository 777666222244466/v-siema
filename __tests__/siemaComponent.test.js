import { Siema } from '../src';
import Vue from 'vue/dist/vue.js';

describe('Siema', () => {
    Vue.component('siema', Siema);

    beforeEach(() => {
        document.body.innerHTML = `
            <div id="app">
                <siema
                    ref="siema"
                    :loop="true"
                >
                    <div>Slide 1</div>
                    <div>Slide 2</div>
                    <div>Slide 3</div>
                    <div>Slide 4</div>
                    <div>Slide 5</div>
                </siema>
            </div>
        `;
    });

    it('can mount', async () => {
        await createVm();

        expect(document.body.innerHTML).toMatchSnapshot();
    });
});

async function createVm() {
    const vm = new Vue({
        el: '#app',

        data() {
            return {
                slides: [
                    {
                        title: 'Slide 1',
                    },
                    {
                        title: 'Slide 2',
                    },
                    {
                        title: 'Slide 3',
                    },
                ],
            };
        },
    });

    await Vue.nextTick(() => {});

    return { app: vm, component: vm.$children[0] };
}
