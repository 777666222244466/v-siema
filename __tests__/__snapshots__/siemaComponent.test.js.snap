// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Siema can mount 1`] = `
"
            <div id=\\"app\\"><div class=\\"siema\\" style=\\"overflow: hidden; direction: ltr; cursor: -webkit-grab;\\"><div style=\\"width: 0px; -webkit-transition: all 500ms ease;\\"><div style=\\"float: left; width: 14.285714285714286%;\\"><div>Slide 5</div></div><div style=\\"float: left; width: 14.285714285714286%;\\"><div>Slide 1</div></div><div style=\\"float: left; width: 14.285714285714286%;\\"><div>Slide 2</div></div><div style=\\"float: left; width: 14.285714285714286%;\\"><div>Slide 3</div></div><div style=\\"float: left; width: 14.285714285714286%;\\"><div>Slide 4</div></div><div style=\\"float: left; width: 14.285714285714286%;\\"><div>Slide 5</div></div><div style=\\"float: left; width: 14.285714285714286%;\\"><div>Slide 1</div></div></div></div></div>
        "
`;
